haystack/nodes/retriever/base.py

```diff
@@ -363,12 +362,6 @@ def run_query_batch(
         return output, "output_1"
 
     def run_indexing(self, documents: List[Union[dict, Document]]):
-        if self.__class__.__name__ in ["DensePassageRetriever", "EmbeddingRetriever"]:
-            documents = deepcopy(documents)
-            document_objects = [Document.from_dict(doc) if isinstance(doc, dict) else doc for doc in documents]
-            embeddings = self.embed_documents(document_objects)  # type: ignore
-            for doc, emb in zip(document_objects, embeddings):
-                doc.embedding = emb
```

Comment:
Great! This removes the need for https://github.com/deepset-ai/haystack/pull/3243 :star_struck: